rename_images_by_date.py

```diff
@@ -42,68 +44,123 @@ def findDuplicates(file_dict):
+   time_delay = datetime.timedelta(hours=int(time_split[0]),minutes=int(time_split[1]),seconds=int(time_split[2]))
+   print("time: " + str(time_delay))
+if cam_name is not None:
+   print("camera name: " + cam_name)
+dir = args.Dir 
+print("target directory: " + dir)
 files = {}
 
+#Find 'valid' file to rename
 for filename in sorted(os.listdir(dir)):
+    if isVerbose:
+        print("File found: " + filename)
 	#check for images in the directory
-	if filename.endswith(".jpg") or filename.endswith(".JPG") or filename.endswith(".jpeg") or filename.endswith(".JPEG") or filename.endswith(".PNG")or filename.endswith(".png"):
-		date = get_date_taken(dir+"/"+filename)
-		if date:
-			files[filename] = date
+    if filename.endswith(".jpg") or filename.endswith(".JPG") or filename.endswith(".jpeg") or filename.endswith(".JPEG") or filename.endswith(".PNG")or filename.endswith(".png"):
+        date = get_date_taken(dir+"/"+filename)
+        if date and useDate:
+            files[filename] = date
+        else:
+            files[filename] = filename
 
-sorted_files  = sorted(files.items(), key=lambda kv: kv[1])
+if isVerbose:
+    print("Found {} file(s) to rename.".format(len(files)))
 
-files = {}
+if len(files) == 0:
+    print("No file found. Aborted.")
+    sys.exit()
+sorted_files  = sorted(files.items(), key=lambda kv: kv[1])
 
+#Rename 
 for filename, date in sorted_files:
-	_date_,_time_ = date.split(" ")
-	date_split = _date_.split(":")
-	time_split = _time_.split(":")
-	date_struct = datetime.datetime(year=int(date_split[0]), month=int(date_split[1]), day=int(date_split[2]), hour=int(time_split[0]), minute=int(time_split[1]),second=int(time_split[2]))
+    if useDate:
+	    _date_,_time_ = date.split(" ")
+	    date_split = _date_.split(":")
+	    time_split = _time_.split(":")
+	    date_struct = datetime.datetime(year=int(date_split[0]), month=int(date_split[1]), day=int(date_split[2]), hour=int(time_split[0]), minute=int(time_split[1]),second=int(time_split[2]))
 	
-	if(cam_name == get_cam_model(dir+"/"+filename)):
-		date_struct = date_struct + time_delay
-	else:
-		print((cam_name,get_cam_model(dir+"/"+filename)))
+	    if(cam_name == get_cam_model(dir+"/"+filename)):
+		    date_struct = date_struct + time_delay
+	    else:
+		    print((cam_name,get_cam_model(dir+"/"+filename)))
+
+	    new_filename = str(date_struct.year)+"_"+str(date_struct.month).zfill(2)+"_"+str(date_struct.day).zfill(2)+"_"+str(date_struct.hour).zfill(2)+"_"+str(date_struct.minute).zfill(2)+"_"+str(date_struct.second).zfill(2)+filename[filename.rfind("."):]
+	    files[filename] = new_filename
+    
+    if prefix != "":
+        previousname = files[filename]
+        new_filename = prefix + "_" + previousname
+        files[filename] = new_filename
+        if isVerbose:
+            print("Prefix added: " + filename + " -> " + new_filename)
+    if suffix != "":
+        previousname,ext = files[filename].split(".")
```

Comment:
previous_name ... filename.split

Reply:
No. I want the latest modified name and to add the suffix to this one. For an example, if I have used prefix first, I want to add the suffix to the name containing already the prefix